integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -58,6 +57,8 @@ object CarbonPreInsertionCasts extends Rule[LogicalPlan] {
         )
     }
     if (child.output.size >= relation.carbonRelation.output.size) {
+      sparkVersion21 = !CarbonClassReflectionUtils.hasField("query", InsertIntoCarbonTable)
```

Comment:
Don't check the version depends on field, please check spark version

Reply:
In some places the SparkContext or SparkSession is directly not available. Therefore used the approach of accessing the field and if not present the assume it as other version.

Reply:
I think we can add a parameter called "SPARK_VERSION" into CarbonProperties, when start CarbonSession, set this parameter as SparkContext.version, and then we can use this parameter in any places